### DISCUSSION TITLE
Example for powershell extension

### DISCUSSION THREAD

**Discussion by cz-dev-ge** [wixtoolset/issues#7417] [Questions] [answered]
Hi there,

I am upgrading a WiX3 project to WiX 4. My `wxs` contains the old `http://schemas.gardiner.net.au/PowerShellWixExtensionSchema` and a script like this:

```xml
<powershell:Script Id="MyScript" Elevated="yes">
  <![CDATA[echo Hello World!]]>
</powershell:Script>
```

This is not available in WiX4. But there is `http://wixtoolset.org/schemas/v4/wxs/powershell`.

Unfortunately I could not find any examples on how to use it. Could someone please provide one?

Best regards.

**Comment by robmen** [comment, marked as answer]
[PowerShell docs are here](https://wixtoolset.org/docs/schema/powershell/). It is designed to install (old) PowerShell stuff. Not sure it is even useful in modern PowerShell.

**Reply by cz-dev-ge** [reply]
Thanks. Whats the intended way to call scripts?

**Reply by robmen** [reply]
There is no such intention in the Windows Installer nor WiX itself today. If you must write a custom action, I personally recommend writing data-driven custom actions that integrate with the Windows Installer transaction to ensure proper install, uninstall, upgrade, and patch behavior with rollback for each.

**Reply by cz-dev-ge** [reply]
Thank you. I'll give that a shot.
